src/main/java/io/cdap/http/internal/ChannelChunkResponder.java

```diff
@@ -54,14 +59,35 @@ public void sendChunk(ByteBuf chunk) throws IOException {
     if (!channel.isActive()) {
       throw new IOException("Connection already closed.");
     }
+    int chunkSize = chunk.readableBytes();
     channel.write(new DefaultHttpContent(chunk));
+    tryFlush(chunkSize);
   }
 
   @Override
-  public void close() throws IOException {
+  public void flush() {
+    // Use the limit as the size to force a flush
+    tryFlush(chunkMemoryLimit);
```

Comment:
Haven't looked at the cases where flush() gets invoked. But if user calls flush() periodically,  passing chunkMemoryLimit would break the assumption of bufferedSize overestimating memory usage. 

For instance, user writes code like 
while(numOfChunks) {
  sendChunk(smallChunk); // where chunk is just a few bytes.
  flush();
}
//  bufferedSize  would be negative by now
// Then later
while(numOfChunks) {
  sendChunk(largeChunk); // potentially oom due to bufferedsize being negative before the whileloop
}

Reply:
We do `long newSize = bufferedSize.addAndGet(size);` and then `bufferedSize.addAndGet(-1 * newSize);`. It shouldn't become negative.